### ISSUE TITLE
How can a local pdf be loaded properly?

### ISSUE THREAD

**Issue by adambeer** [stephanrauh/ngx-extended-pdf-viewer#870]
The viewer code is hosted on a web server while the pdf wishing to be loaded is on a users phone. Is it possible to load the locally saved pdf into the remotely hosted viewer? If its possible, does the process involve "uploading" the pdf or is loading done locally?

Edit: just for clarity, the pdf on the phone is saved in a place that a specific app can link to it.

**Comment by stephanrauh** [comment]
I've just checked my iPad. I've managed to upload a PDF file using the "file open" button. You don't have programmatic access to the local file system, but iOS allows the user to access several folders. Maybe the folder you need is among the allowed folders.

**Comment by adambeer** [comment]
I saw the open file functionality. I feel like this is as simple as setting the scr to use a file:// url just like you would do on any other device. Im guessing id have to programmatically replicate the open file functionality using some sort of local url?

**Comment by stephanrauh** [comment]
I'm afraid that's not possible. Browsers put JavaScript in a sandbox, for very good reasons. But for us programmers that's just infuriating. AFAIK the only way to escape the sandbox is using frameworks like Ionic and Capacitor.

**Comment by adambeer** [comment]
So in theory, if I were to be able to get the viewer running locally on a users device (say through a react-native webview) accessing local PDF's using file:/// would work?

**Comment by stephanrauh** [comment]
Yes. ngx-extended-pdf-viewer can't do that, because I've focus too much on web applications. But the base library, pdf.js, offers limited support for node.js. At least two files of the library deal specifically with a node.js environment.

I'm afraid I don't know how to create a useful pdf viewer running on node.js, but maybe this pointer helps you:

https://github.com/mozilla/pdf.js/tree/master/examples/node/pdf2png

**Comment by adambeer** [comment]
I actually successfully got the viewer working in a react-native webview yesterday! It has the option to directly load a local html file. The only thing I had to do was move all js files inline (including the pdf.js support files). I have yet to get a local file loaded yet but thats step 2.

My next challenge with this library is configuring it to not call methods like loadViewer and loadPDF as I am loading them inline, and to learn how to automate packing all the js files into inline script inside the html file automatically. The index turns into a 165k line file doing this manually...
